extension/content/components/common/AppSidebar.tsx

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import { NavLink } from "react-router-dom";
+import { Icon, Nav, Sidenav, Popover, Whisper } from "rsuite";
+
+import { useEnvironmentState } from "devtools/contexts/environment";
+import { splitCamelCase } from "devtools/utils/helpers";
+
+export default function AppSidebar(): JSX.Element {
```

Comment:
I used `JSX.Element` because that is what Typescript inferred the return value of this function was. `JSX.Element` is the type that you get from JSX literals, and specifically, it's the *return value* of this function. `React.FC` (short for `React.FunctionComponent`) is the type of a functional component, and it's the type of this expression:

```
const AppSideBar: React.FC = (props) => { /* some jsx */ }
```

I also learned that `JSX.Element` is an alias for `React.ReactElement` in our project (it would be different in a project that uses JSX but not React).

So I think it depends on how you define the component. If it's `function Foo() {}` then it should be `JSX.Element` or `React.ReactElement`. If it's `const Foo = () => {}`, then it just be `React.FC`.